twosixlabs/armory
# --check mode for armory run

Fixes #382 

Can now quickly test code in a scenario by using the CLI flag `--check`. This works by essentially setting all loaded datasets to have a total size equal to a single batch (note that this means that if the batch size is larger, it will take longer). This required skipping the filtering defense portion of the poisoning scenario, but otherwise works fine on all scenarios.

To test:
```
python -m armory scenario_configs/mnist_baseline.json --check
python -m armory scenario_configs/mnist_baseline.json --check --no-docker
```

I also ensure that the `--debug` flag is carried into the scenario from the calling function, which partially addresses #500 .
